src/averageAges.js

```diff
@@ -55,7 +69,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredPeople = onlyWithSon
+    ? people.filter(({ sex, mother }) => (
+      people.find(({ name }) => mother === name)) && sex === 'm')
+    : people.filter(({ mother }) => (
+      people.find(({ name }) => mother === name)));
+  const length = filteredPeople.length;
+
+  return ((filteredPeople.reduce((sumAges, { born, mother }) => (
+    sumAges + born - people.find(({ name }) => mother === name).born), 0))
+    / length);
```

Comment:
Better create variable `motherBorn` for example, this way code would be easier to read.